core/src/test/java/org/apache/spark/ExecutorPluginSuite.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark;
+
+import org.apache.spark.api.java.JavaSparkContext;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+// Tests loading plugins into executors
+public class ExecutorPluginSuite {
+  // Static value modified by testing plugin to ensure plugin loaded correctly.
+  public static int numSuccessfulPlugins = 0;
+
+  private SparkConf initializeSparkConf(String pluginNames) {
+    return new SparkConf()
+        .setMaster("local")
+        .setAppName("test")
+        .set("spark.executor.plugins", pluginNames);
```

Comment:
Use `.set(CONSTANT, Seq(...))`

Reply:
Sorry, could you explain this for me? Do you mean instead of `.set("spark.executor.plugins", pluginNames)` I should a) have a variable storing the config name and b) pass the plugin names to initializeSparkConf as a list, and have it as `.set(EXECUTOR_PLUGIN_CONF_NAME, String.join(",", pluginNames)`?

Reply:
`.set(EXECUTOR_PLUGINS, Seq("class1", "class2"))`

Reply:
Is it alright to use Seq in a Java file? I couldn't find it in any other java files in Spark that aren't in a target/ directory. I don't think the compiler is getting mad when I import it but want to make sure I'm sticking with the usual style rules

Reply:
I had the same first response. The original executor plugin code was a class in Java, so I figured I should keep my test in Java as well (manually checked beforehand, couldn't find any instances of Java stuff being primarily tested in Scala so I figured it best not to do that). Perhaps it's for the plugin-writers to get to choose their plugin language, since it's easier to extend a Java interface in Scala, than to extend a Scala trait in Java? I'll leave it as is for the time being

Reply:
yeah I made the interface in java to make it easier for plugin-writers, since it didn't need scala at all.  actually an empty trait also compiles to an empty interface in java with no references to scala, but I figured this would help remember that if the interface gets added to in the future.

I don't see any reason the test needs to be in java, but don't feel strongly either.